src/store/useProjectStore.ts

```diff
@@ -20,30 +20,50 @@ const defaultState: Omit<
 	| "deleteProject"
 	| "createProjectFromManifest"
 	| "setEditorWidth"
-	| "setLatestOpenedTab"
 	| "setPendingFile"
+	| "setLatestOpened"
 > = {
 	projectsList: [],
-	latestOpenedTab: "",
 	isLoadingProjectsList: true,
 	initialEditorWidth: 50,
+	currentProjectId: undefined,
 	pendingFile: undefined,
+	latestOpened: {
+		tab: "",
+		deploymentId: "",
+		sessionId: "",
+		projectId: undefined,
+	},
 };
 
 const store: StateCreator<ProjectStore> = (set, get) => ({
 	...defaultState,
 
-	setEditorWidth: (width) => {
+	setLatestOpened: (type, value, projectId) => {
 		set((state) => {
-			state.initialEditorWidth = width;
+			if (projectId && projectId !== state.latestOpened.projectId) {
+				state.latestOpened = {
+					tab: type === "tab" ? value : "",
+					deploymentId: type === "deploymentId" ? value : "",
+					sessionId: type === "sessionId" ? value : "",
```

Comment:
I don't understand why we need to save `deploymentId` and `sessionId`, what will this give us??